File: BigFootVentures/BigFootVentures.Business.Objects/Management/Company.cs
using System;

namespace BigFootVentures.Business.Objects.Management
{
    public sealed class Company : BusinessBase
    {
        #region "Private Members"

        private string _deletionRequest { get; set; }

        #endregion

        #region "Properties"

        public string AccountRecordType { get; set; }
        public string AccountOwner { get; set; }

        public Company ParentAccount { get; set; }

        public string CompanyName { get; set; }
        public string FormerName { get; set; }

        public string LastName { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string Suffix { get; set; }
        public string Title { get; set; }
        public string Salutation { get; set; }

        public string Type { get; set; }
        public string Description { get; set; }

        public string Phone { get; set; }
        public string Mobile { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }

        public string Industry { get; set; }
        public string Employees { get; set; }

        public string NameID { get; set; }
        public string OfficeIDOHIM { get; set; }
        public string OfficeIDGB { get; set; }
        public string OHIMNumTrademarks { get; set; }
        public string OHIMNUMOppositions { get; set; }
        public string OHIMOwnerID { get; set; }
        public string AddressType { get; set; }
        public string CompanySize { get; set; }
        public bool EscrowAgent { get; set; }
        public bool Broker { get; set; }

        public string DeletionRequest { get { return this._deletionRequest; } set { this._deletionRequest = value; this.DeletionRequestChk = string.Equals(value, true.ToString(), StringComparison.InvariantCultureIgnoreCase); } }
        public string DeletionRequestReason { get; set; }

        public string CompanyRegistrationNumber { get; set; }
        public string CountryOfIncorporation { get; set; }
        public string DateOfIncorporation { get; set; }
        public string TaxNumber { get; set; }
        public string Officers { get; set; }

        public string TMFilingCost { get; set; }
        public string TMCancellationCost { get; set; }
        public string TMOppositionCost { get; set; }
        public string TMPriorityCost { get; set; }
        public string TMRegistrationCertificateCost { get; set; }
        public string TMResearchCost { get; set; }
        public string OtherCosts { get; set; }

        public bool BigFootAccredited { get; set; }
        public bool BigFootGroup { get; set; }

        public string ShippingCountry { get; set; }
        public string ShippingStreet { get; set; }
        public string ShippingCity { get; set; }
        public string ShippingState { get; set; }
        public string ShippingPostalCode { get; set; }

        public UserAccount AssignedStaff { get; set; }
        public UserAccount AssignedSupervisor { get; set; }

        public string DueDate { get; set; }

        #endregion

        #region "Calculated Properties"

        public bool DeletionRequestChk { get; set; }

        #endregion
    }
}

File: BigFootVentures/BigFootVentures.Application.Web/Scripts/Views/Shared/_Components/Modals/AdvanceSearch/_Company.js
function BuildCompanyQuery(query) {
    var name = $('[name="Input.Filter.Company.Name"]').val();
    var accountRecordType = $('[name="Select.Filter.Company.AccountRecordType"]').val();
    var formerName = $('[name="Input.Filter.Company.FormerName"]').val();
    var type = $('[name="Select.Filter.Company.Type"]').val();
    var parentAccount = $('[name="Input.Filter.Company.ParentAccount"]').val();
    var phone = $('[name="Input.Filter.Company.Phone"]').val();
    var fax = $('[name="Input.Filter.Company.Fax"]').val();
    var email = $('[name="Input.Filter.Company.Email"]').val();
    var companySize = $('[name="Select.Filter.Company.CompanySize"]').val();
    var industry = $('[name="Select.Filter.Company.Industry"]').val();
    var nameID = $('[name="Input.Filter.Company.Employees"]').val();
    var employees = $('[name="Input.Filter.Company.NameID"]').val();
    var officeIDGB = $('[name="Input.Filter.Company.OfficeIDGB"]').val();
    var OHIMNumOppositions = $('[name="Input.Filter.Company.OHIMNumOppositions"]').val();
    var escrowAgent = $('[name="Select.Filter.Company.EscrowAgent"]').val();
    var broker = $('[name="Select.Filter.Company.Broker"]').val();
    var companyRegistrationNumber = $('[name="Input.Filter.Company.CompanyRegistrationNumber"]').val();
    var countryOfIncorporation = $('[name="Select.Filter.Company.CountryOfIncorporation"]').val();
    var officers = $('[name="Input.Filter.Company.Officers"]').val();
    var taxNumber = $('[name="Input.Filter.Company.TaxNumber"]').val();
    var bigFootAccredited = $('[name="Select.Filter.Company.BigFootAccredited"]').val();
    var shippingCountry = $('[name="Select.Filter.Company.ShippingCountry"]').val();

    if (name != null && name != undefined && name.length > 0) {
        query += '&name=' + name;
    }

    if (accountRecordType != '') {
        query += '&accountRecordType=' + accountRecordType;
    }

    if (formerName != null && formerName != undefined && formerName.length > 0) {
        query += '&formerName=' + formerName;
    }

    if (type != '') {
        query += '&type=' + type;
    }

    if (parentAccount != null && parentAccount != undefined && parentAccount.length > 0) {
        query += '&parentAccount=' + parentAccount;
    }

    if (phone != null && phone != undefined && phone.length > 0) {
        query += '&phone=' + phone;
    }

    if (fax != null && fax != undefined && fax.length > 0) {
        query += '&fax=' + fax;
    }

    if (email != null && email != undefined && email.length > 0) {
        query += '&email=' + email;
    }

    if (companySize != '') {
        query += '&companySize=' + companySize;
    }

    if (industry != '') {
        query += '&industry=' + industry;
    }

    if (nameID != null && nameID != undefined && nameID.length > 0) {
        query += '&nameID=' + nameID;
    }

    if (employees != null && employees != undefined && employees.length > 0) {
        query += '&employees=' + employees;
    }

    if (officeIDGB != null && officeIDGB != undefined && officeIDGB.length > 0) {
        query += '&officeIDGB=' + officeIDGB;
    }

    if (OHIMNumOppositions != null && OHIMNumOppositions != undefined && OHIMNumOppositions.length > 0) {
        query += '&OHIMNumOppositions=' + OHIMNumOppositions;
    }

    if (escrowAgent != 'NotSet') {
        query += '&escrowAgent=' + escrowAgent;
    }

    if (broker != 'NotSet') {
        query += '&broker=' + broker;
    }

    if (companyRegistrationNumber != null && companyRegistrationNumber != undefined && companyRegistrationNumber.length > 0) {
        query += '&companyRegistrationNumber=' + companyRegistrationNumber;
    }

    if (countryOfIncorporation != '') {
        query += '&countryOfIncorporation=' + countryOfIncorporation;
    }

    if (officers != null && officers != undefined && officers.length > 0) {
        query += '&officers=' + officers;
    }

    if (taxNumber != null && taxNumber != undefined && taxNumber.length > 0) {
        query += '&taxNumber=' + taxNumber;
    }

    if (bigFootAccredited != 'NotSet') {
        query += '&bigFootAccredited=' + bigFootAccredited;
    }

    if (shippingCountry != '') {
        query += '&shippingCountry=' + shippingCountry;
    }

    return query;
}